### PR TITLE
C20 Tayam, Luminous Enigma

### PR DESCRIPTION
**Pull request by htrajan**

### CODE REVIEW

**Comment by htrajan** [comment]
Edge case not yet working: A creature has multiple types of counters on it. You cannot pay more than 1 of the counter cost using that creature unless you strictly remove multiple counters of the same type.

File: Mage.Sets/src/mage/cards/t/TayamLuminousEnigma.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new TayamLuminousEnigmaReplacementEffect()));
+
+        // {3}, Remove three counters from among creatures you control: Put the top three cards of your library into your graveyard, then return a permanent card with converted mana cost 3 or less from your graveyard to the battlefield.
+        PutTopCardOfLibraryIntoGraveControllerEffect millEffect = new PutTopCardOfLibraryIntoGraveControllerEffect(3);
+        millEffect.concatBy(".");
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, millEffect, new GenericManaCost(3));
+        ability.addCost(new TayamLuminousEnigmaCost());
+        TayamLuminousEnigmaEffect effect = new TayamLuminousEnigmaEffect();
+        effect.setText(", then return a permanent card with converted mana cost 3 or less from your graveyard to the battlefield");
+        ability.addEffect(effect);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/t/TayamLuminousEnigma.java:65]: Did you check if the text comes out correctly?

File: Mage.Sets/src/mage/cards/t/TayamLuminousEnigma.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+class TayamLuminousEnigmaEffect extends OneShotEffect {
+
+    private final FilterCard filter;
+
+    TayamLuminousEnigmaEffect() {
+        super(Outcome.Benefit);
+        this.filter = new FilterPermanentCard("permanent card in your graveyard with converted mana cost 3 or less");
+        this.filter.add(new ConvertedManaCostPredicate(ComparisonType.FEWER_THAN, 4));
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/t/TayamLuminousEnigma.java:170]: This filter can be private static final

File: Mage.Sets/src/mage/cards/t/TayamLuminousEnigma.java (hunk 1 above)
Comment (htrajan) [reply Mage.Sets/src/mage/cards/t/TayamLuminousEnigma.java:65]: Yep!

image

File: Mage.Sets/src/mage/cards/t/TayamLuminousEnigma.java (hunk 2 above)
Comment (htrajan) [reply Mage.Sets/src/mage/cards/t/TayamLuminousEnigma.java]: fixed

**Pull request merged**